src/rtp/srtp.rs

```diff
@@ -389,134 +582,382 @@ impl Derived {
+        fn rtp_hmac(&self, buf: &mut [u8], srtp_index: u64, hmac_start: usize) {
+            let sha1 = self.clone();
 
-impl RtpHmac for Sha1 {
-    fn rtp_hmac(&self, buf: &mut [u8], srtp_index: u64, hmac_start: usize) {
-        let sha1 = self.clone();
+            let roc = (srtp_index >> 16) as u32;
 
-        let roc = (srtp_index >> 16) as u32;
+            let tag = sha1.hmac(&[&buf[..hmac_start], &roc.to_be_bytes()]);
 
-        let tag = sha1.hmac(&[&buf[..hmac_start], &roc.to_be_bytes()]);
+            buf[hmac_start..(hmac_start + HMAC_TAG_LEN)].copy_from_slice(&tag[0..HMAC_TAG_LEN]);
+        }
 
-        buf[hmac_start..(hmac_start + SRTP_HMAC_LEN)].copy_from_slice(&tag[0..SRTP_HMAC_LEN]);
-    }
+        fn rtp_verify(&self, buf: &[u8], srtp_index: u64, cmp: &[u8]) -> bool {
+            let sha1 = self.clone();
 
-    fn rtp_verify(&self, buf: &[u8], srtp_index: u64, cmp: &[u8]) -> bool {
-        let sha1 = self.clone();
+            let roc = (srtp_index >> 16) as u32;
 
-        let roc = (srtp_index >> 16) as u32;
+            let tag = sha1.hmac(&[buf, &roc.to_be_bytes()]);
 
-        let tag = sha1.hmac(&[buf, &roc.to_be_bytes()]);
+            &tag[0..HMAC_TAG_LEN] == cmp
+        }
 
-        &tag[0..SRTP_HMAC_LEN] == cmp
-    }
+        fn rtcp_hmac(&self, buf: &mut [u8], hmac_index: usize) {
+            let sha1 = self.clone();
 
-    fn rtcp_hmac(&self, buf: &mut [u8], hmac_index: usize) {
-        let sha1 = self.clone();
+            let tag = sha1.hmac(&[&buf[0..hmac_index]]);
 
-        let tag = sha1.hmac(&[&buf[0..hmac_index]]);
+            buf[hmac_index..(hmac_index + HMAC_TAG_LEN)].copy_from_slice(&tag[0..HMAC_TAG_LEN]);
+        }
 
-        buf[hmac_index..(hmac_index + SRTP_HMAC_LEN)].copy_from_slice(&tag[0..SRTP_HMAC_LEN]);
-    }
+        fn rtcp_verify(&self, buf: &[u8], cmp: &[u8]) -> bool {
+            let sha1 = self.clone();
 
-    fn rtcp_verify(&self, buf: &[u8], cmp: &[u8]) -> bool {
-        let sha1 = self.clone();
+            let tag = sha1.hmac(&[buf]);
 
-        let tag = sha1.hmac(&[buf]);
+            &tag[0..HMAC_TAG_LEN] == cmp
+        }
+    }
 
-        &tag[0..SRTP_HMAC_LEN] == cmp
+    pub(super) trait ToRtpIv {
+        fn rtp_iv(&self, ssrc: u32, srtp_index: u64) -> RtpIv;
     }
-}
 
-trait ToRtpIv {
-    fn rtp_iv(&self, ssrc: u32, srtp_index: u64) -> RtpIv;
-}
+    impl ToRtpIv for RtpSalt {
+        fn rtp_iv(&self, ssrc: u32, srtp_index: u64) -> RtpIv {
+            let mut iv = [0; 16];
 
-impl ToRtpIv for RtpSalt {
-    fn rtp_iv(&self, ssrc: u32, srtp_index: u64) -> RtpIv {
-        let mut iv = [0; 16];
+            let ssrc_be = ssrc.to_be_bytes();
+            let srtp_be = srtp_index.to_be_bytes();
 
-        let ssrc_be = ssrc.to_be_bytes();
-        let srtp_be = srtp_index.to_be_bytes();
+            iv[4..8].copy_from_slice(&ssrc_be);
 
-        iv[4..8].copy_from_slice(&ssrc_be);
+            for i in 0..8 {
+                iv[i + 6] ^= srtp_be[i];
```

Comment:
These don't need to be `^=` and then could use `copy_from_slice`. I didn't change it in this PR because I didn't want to touch the existing profile's implementation